pkg/slo-controller/noderesource/noderesource.go

```diff
@@ -195,6 +199,88 @@ func (r *NodeResourceReconciler) updateGPUNodeResource(node *corev1.Node, device
 	return err
 }
 
+func (r *NodeResourceReconciler) updateGPUDriverAndModel(node *corev1.Node, device *schedulingv1alpha1.Device) error {
+	if device == nil {
+		return nil
+	}
+
+	hasGPUDevice := false
+	for _, device := range device.Spec.Devices {
+		if device.Type != schedulingv1alpha1.GPU {
+			continue
+		}
+		hasGPUDevice = true
+	}
+
+	if !hasGPUDevice {
+		return nil
+	}
+
+	modelList, driverVersion, err := getGPUModelAndDriverInfo()
+	if err != nil {
+		klog.Errorf("failed to get GPU Model and driver info: %v", err)
+		return err
+	}
+
+	model := ""
+	for i, v := range modelList {
+		if v == nil {
+			return fmt.Errorf("get gpu model error")
+		} else if i == 0 {
+			model = *v
+		} else if model != *v {
+			return fmt.Errorf("node contains multi-gpu, model: %v, %v", model, v)
+		}
+	}
+
+	// NVIDIA Driver 470 report GPU Model with "NVIDIA " Prefix
+	model = strings.TrimPrefix(model, "NVIDIA ")
+
+	// A100 SXM4 80GB -> A100-SXM4-80GB
+	// Tesla P100-PCIE-16GB -> Tesla-P100-PCIE-16GB
+	// Tesla V100-SXM2-16GB -> Tesla-V100-SXM2-16GB
+	// Tesla T4 -> Tesla-T4
+	// Tesla P40 -> Tesla-P40
+	// Tesla M40 -> Tesla-M40
+	// GeForce RTX 2080 Ti -> GeForce-RTX-2080-Ti
+	// GeForce GTX 1080 Ti -> GeForce-GTX-1080-Ti
+	transModel := strings.ReplaceAll(model, " ", "-")
+
+	copyNode := node.DeepCopy()
+	if copyNode.Labels == nil {
+		copyNode.Labels = make(map[string]string)
+	}
+	copyNode.Labels[extension.GpuModel] = transModel
+	copyNode.Labels[extension.GpuDriver] = driverVersion
+
+	patch := client.MergeFrom(node)
+	if err = r.Client.Patch(context.Background(), copyNode, patch); err != nil {
+		klog.Errorf("patch node gpu model and version fail, err:%v", err)
+		return err
+	}
+	return nil
+}
+
+func getGPUModelAndDriverInfo() ([]*string, string, error) {
+	var modelList []*string
+	driverVersion, err := nvmlbindings.GetDriverVersion()
```

Comment:
hmm.. Does it work?  slo-controller just a add-on controller works in the cluster not the node.

Reply:
fixed